table/tables/partition_test.go

```diff
@@ -530,3 +530,19 @@ func (ts *testSuite) TestIssue21574(c *C) {
 	tk.MustExec("drop table t_21574")
 	tk.MustExec("create table t_21574 (`key` int, `table` int) partition by list columns (`key`,`table`) (partition p0 values in ((1,1)));")
 }
+
+func (ts *testSuite) TestIssue24746(c *C) {
+	tk := testkit.NewTestKitWithInit(c, ts.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop tables if exists t_24746")
+	tk.MustExec("create table t_24746 (a int, b varchar(60), c int, primary key(a)) partition by range(a) (partition p0 values less than (5),partition p1 values less than (10), partition p2 values less than maxvalue)")
+	defer tk.MustExec("drop table t_24746")
+	_, err := tk.Exec("insert into t_24746 partition (p1) values(4,'ERROR, not matching partition p1',4)")
```

Comment:
Better to use `tk.ExecToErr` here otherwise we have to close the `res` in case of resource leak~

Reply:
Fixed, thanks for the suggestion.